pygeo/parameterization/DVGeo.py

```diff
@@ -1858,14 +1890,22 @@ def convertSensitivityToDict(self, dIdx, out1D=False):
 
         # Add in child portion
         for iChild in range(len(self.children)):
-            childdIdx = self.children[iChild].convertSensitivityToDict(dIdx, out1D=out1D)
+            childdIdx = self.children[iChild].convertSensitivityToDict(dIdx, out1D=out1D, compName=compName)
             # update the total sensitivities with the derivatives from the child
             for key in childdIdx:
                 if key in dIdxDict.keys():
                     dIdxDict[key] += childdIdx[key]
                 else:
                     dIdxDict[key] = childdIdx[key]
 
+        # replace other names with user
+        if compName and self.useComposite:
+            array = []
+            for key, val in dIdxDict.items():
+                array.append(val)
+            array = np.hstack(array)
+            dIdxDict = {self.DVComposite.name: array}
```

Comment:
If I get this right, here you are just renaming the key of the dictionary, while the actual mapping happens in `mapSensToComp`? basically when using a composite variable we are still populating the dictionary as we have regular DVs, but the actual entries are not related to the specific DV anymore after the mapping.

Reply:
The function `totalSensitivity` computes `dIdx`, which is the array of sensitivities wrt either the original DVs or the compDVs depending on useComposite. This function just converts that array to a dictionary populated by DV keys, and of course we just put one DVgroup in the dictionary when compDV is activated.